src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1003,6 +1010,50 @@ public bool ProcessTransaction(Transaction transaction, int? blockHeight = null,
 
             lock (this.lockObject)
             {
+                // Do a pre-scan of the incoming transaction's inputs to see if they're used in other wallet transactions already.
+                foreach (TxIn input in transaction.Inputs)
+                {
+                    // See if this input is being used by another wallet transaction present in the index.
+                    // The inputs themselves may not belong to the wallet, but the transaction data in the index has to be for a wallet transaction.
```

Comment:
Seems we only use the trx hash (uint256) I dont think we need to store this type `WalletIndexData` at all just reference the trx hash

Reply:
I do see the benefit of indexing the whole structure though